### PR TITLE
Chronos: Add a more stable asym loss for forecaster training

### PR DESCRIPTION
**Pull request by TheaperDeng**
## Description
This `AsymWeightLoss` is similar to `LinexLoss` we involved in https://github.com/intel-analytics/BigDL/pull/4779 and demo in https://github.com/intel-analytics/BigDL/pull/5627 but with following pros:

- More stable, it could handle different models and bring better underestimation rate for different seed
- Reduce Nan problem.

### 1. Why the change?

- More stable, it could handle different models and bring better underestimation rate for different seed
- Reduce Nan problem.

### 2. User API changes

```python
from bigdl.chronos.pytorch.loss import AsymWeightLoss
AsymWeightLoss(underestimation_penalty=10)  # > 1 to penalize underestimation and < 1 to penalize overestimation
```

### 3. Summary of the change 

- add this loss and uts
- change the demo notebook to use this loss

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by TheaperDeng** [comment]
@yangqing-yq please have a look either since this may make the asymmetric loss more stable and perform better.

File: python/chronos/test/bigdl/chronos/pytorch/loss/test_asym_wight.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def tearDown(self):
+        pass
+
+    def test_linex_loss(self):
+        y = torch.rand(100, 10, 2)
+        yhat_high = y + 1
+        yhat_low = y - 1
+
+        # when a is set to > 0, this loss panelize underestimate more.
```
Comment (plusbang) [inline python/chronos/test/bigdl/chronos/pytorch/loss/test_asym_wight.py:41]: The comments of line 41 and line 45 are out of date, other LGTM.

**Review by plusbang — APPROVED** [review]

File: python/chronos/src/bigdl/chronos/pytorch/loss/asym_weight.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from bigdl.nano.utils.common import invalidInputError
+
+
+class AsymWeightLoss(nn.Module):
+    '''
+    AsynWeightLoss is an asymmetric loss.
```
Comment (jason-dai) [inline python/chronos/src/bigdl/chronos/pytorch/loss/asym_weight.py:25]: AsymWeightLoss is an asymmetric loss

**Comment by yangqing-yq** [comment]
Thanks junwei, I will try this asymmetric loss in our use case within NWAI toolkit.
